### DISCUSSION TITLE
Still need node_modules + gem after compiling?

### DISCUSSION THREAD

**Discussion by ldanielw1** [shakacode/react_on_rails#1643] [General]
I'm new to all of this - so I apologize if this question is rather naive...

But our prod environment is containerized, and part of that build is to call `rake assets:precompile`, which I hear also calls webpack compilation. Is it possible, to save space, for us to get rid of the node_modules after assets are compiled? And is there any possibility that, after that, there might even be the potential to strip out the react_on_rails gem from bundle (so that we don't need to have execjs / node.js installed in the container as well)?

**Comment by justin808** [comment]
yes on removing node modules and maybe all your JS code that got processed by webpack

no on removing the ruby gem

I hope that helps!

**Reply by ldanielw1** [reply]
That's very helpful! And do you know if there'd be any dangers to removing the react_on_rails gem from bundle in prod after precompilation?

**Reply by justin808** [reply]
the gem is not in the bundles produced by webpack

**Comment by ldanielw1** [comment]
Ah sry but could I delete it from the gem bundle?

**Reply by justin808** [reply]
what is the gem bundle?

**Reply by ldanielw1** [reply]
When you run `bundle` and this gem and all of its dependencies get installed locally. Do I still need the gem installed for the runtime? Or is it only needed for build/compile-time?

**Reply by justin808** [reply]
Yes, all gems used by the Rails server are needed at runtime.

**Reply by ldanielw1** [reply]
Got it, so things would break even if I only deleted react_on_rails, and I've precompiled my assets?
